### ISSUE TITLE
Enable/install a Fortran compiler for Mac in the CI

### ISSUE THREAD

**Issue by valassi** [madgraph5/madgraph4gpu#376]
Hi @hageboeck would it be possible to install a Fortran compiler in the CI please?

I have now added a standalone Fortran test in PR #367, this is meant to mimic even more closely what will go on in the MadEvent integration, so it includes a Fortran program that calls two of C++/CUDA components, not only a ME calculation but also a sample (random+rambo). 

I have included this in the basic build, so now it is failing
https://github.com/madgraph5/madgraph4gpu/runs/5090307331?check_suite_focus=true
```
f77 -I. -I../../include -c fcheck_sa.f -o fcheck_sa.o
make: f77: No such file or directory
``` 
(I am using $(FC) in the makefile for the Fortran compiler, just like $(CXX) is the c++ compiler, in case you want to set it up to something else)

By the way, eventually I would like to create a basic test that also checks that the Fortran and c/cuda executables give the same results. I can add that as a target. Anyway, I was wondering, maybe I would rename the 'make check' target that is now used to run the test in the CI - the reason is that 'check.exe' is the name of the executable (well we could change that too, eventually), so I would prefer something a bit more explict like 'make civalidation'. 'make citests' or something similar. Suggestions/feedback?

Thanks! Andrea

**Comment by valassi** [comment]
PS Note that the problem is only on Mac (another issue in #375), as instead on Linux all is ok
https://github.com/madgraph5/madgraph4gpu/runs/5090306909?check_suite_focus=true#step:4:104
```
g++ -o runTest.exe ./CPPProcess.o ./RandomNumberKernels.o ./RamboSamplingKernels.o ./MatrixElementKernels.o ./BridgeKernels.o ./CrossSectionKernels.o ./testxxx.o  ./testmisc.o  ./runTest.o -ldl -pthread -L../../lib -lmg5amc_common -L../../../../../test/googletest/build/lib/ -lgtest -lgtest_main -Wl,-rpath,'$ORIGIN/../../lib'  
f77 -I. -I../../include -c fcheck_sa.f -o fcheck_sa.o
g++  -O3  -std=c++17 -I. -I../../src -I../../../../../tools  -Wall -Wshadow -Wextra -ffast-math   -march=skylake-avx512 -mprefer-vector-width=256  -DMGONGPU_FPTYPE_DOUBLE -DMGONGPU_HAS_NO_CURAND  -fPIC -c fsampler.cc -o fsampler.o
```

**Comment by hageboeck** [comment]
Hello @valassi,

As you found out, the problem is only on Mac. Here, we so far get only what github provides. I'll give it one try to install `gfortran` via homebrew, but we'll have to see whether that's compatible with the rest of the image.

> By the way, eventually I would like to create a basic test that also checks that the Fortran and c/cuda executables give the same results. I can add that as a target. Anyway, I was wondering, maybe I would rename the 'make check' target that is now used to run the test in the CI - the reason is that 'check.exe' is the name of the executable (well we could change that too, eventually), so I would prefer something a bit more explict like 'make civalidation'. 'make citests' or something similar. Suggestions/feedback?

Any name is fine. You just have to sync the Makefile and command that's run in the CI.

**Comment by valassi** [comment]
Thanks Stephan! Lets see how it goes. In the worst case, I will disable the Fortran build/ tests on Mac. This is not really anything people will use for large scale productions on the Grid. But having the software buildable and usable is always handy.

**Comment by hageboeck** [comment]
As per [Github Docs](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.15-Readme.md), one can already invoke it via `gfortran-11`; confirmed in test CI run in #377.

**Comment by valassi** [comment]
Ok thanks. Then what is needed is to propagate this to the Makefile. Can you maybe add something like
```
export FC=gfortran-11
```
in the setup? I think this should be in the setup before the Makefile, rather than in the Makefile.

**Comment by valassi** [comment]
The same issue fixed by Stephan for Mac in #378 also existed for the GPU nodes: FC had to be set to gfortran (the Makefile default is f77). This is now done, the CI for PR #367 is now succeding everywhere (with Fortran enabled) at last.
